aws-rds-dbinstance/aws-rds-dbinstance.json

```diff
@@ -0,0 +1,431 @@
+    },
+    "DeletionProtection": {
+      "type": "boolean",
+      "description": "A value that indicates whether the DB instance has deletion protection enabled. The database can't be deleted when deletion protection is enabled. By default, deletion protection is disabled."
+    },
+    "Domain": {
+      "type": "string",
+      "description": "The Active Directory directory ID to create the DB instance in. Currently, only MySQL, Microsoft SQL Server, Oracle, and PostgreSQL DB instances can be created in an Active Directory Domain."
+    },
+    "DomainIAMRoleName": {
+      "type": "string",
+      "description": "Specify the name of the IAM role to be used when making API calls to the Directory Service."
+    },
+    "EnableCloudwatchLogsExports": {
+      "type": "array",
+      "uniqueItems": true,
+      "insertionOrder": false,
+      "items": {
+        "type": "string"
+      },
+      "description": "The list of log types that need to be enabled for exporting to CloudWatch Logs. The values in the list depend on the DB engine being used."
+    },
+    "EnableIAMDatabaseAuthentication": {
+      "type": "boolean",
+      "description": "A value that indicates whether to enable mapping of AWS Identity and Access Management (IAM) accounts to database accounts. By default, mapping is disabled."
+    },
+    "EnablePerformanceInsights": {
+      "type": "boolean",
+      "description": "A value that indicates whether to enable Performance Insights for the DB instance."
+    },
+    "Engine": {
+      "type": "string",
+      "enum": [
+        "aurora",
+        "aurora-mysql",
+        "aurora-postgresql",
+        "mariadb",
+        "mysql",
+        "oracle-ee",
+        "oracle-se2",
+        "oracle-se1",
+        "oracle-se",
+        "postgres",
+        "postgresql-license",
+        "sqlserver-ee",
+        "sqlserver-se",
+        "sqlserver-ex",
+        "sqlserver-web"
+      ],
+      "description": "The name of the database engine that you want to use for this DB instance."
+    },
+    "EngineVersion": {
+      "type": "string",
+      "description": "The version number of the database engine to use."
+    },
+    "Iops": {
+      "type": "integer",
+      "minimum": 1000,
+      "description": "The number of I/O operations per second (IOPS) that the database provisions. The value must be equal to or greater than 1000."
+    },
+    "KmsKeyId": {
+      "type": "string",
+      "description": "The ARN of the AWS Key Management Service (AWS KMS) master key that's used to encrypt the DB instance."
+    },
+    "LicenseModel": {
+      "type": "string",
+      "enum": [
+        "bring-your-own-license",
+        "general-public-license",
+        "license-included",
+        "postgresql-license"
+      ],
+      "description": "License model information for this DB instance."
+    },
+    "MasterUsername": {
+      "type": "string",
+      "pattern": "^[a-zA-Z][a-zA-Z0-9]{0,15}$",
```

Comment:
This pattern will refuse a valid input in case of these engines

* *Microsoft SQL Server* which length can go up to 128
* *Oracle* length up to 30
* *PostgreSQL* length up to 63

Reply:
Good catch, I must keep this filter broader than WS. To be fixed.